govcd/cse.go

```diff
@@ -0,0 +1,379 @@
+package govcd
+
+import (
+	"encoding/json"
+	"fmt"
+	semver "github.com/hashicorp/go-version"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"github.com/vmware/go-vcloud-director/v2/util"
+	"strings"
+	"time"
+)
+
+// CseCreateKubernetesCluster creates a Kubernetes cluster with the data given as input (CseClusterSettings). If the given
+// timeout is 0, it waits forever for the cluster creation.
+//
+// If the timeout is reached and the cluster is not available (in "provisioned" state), it will return a non-nil CseKubernetesCluster
+// with only the cluster ID and an error. This means that the cluster will be left in VCD in any state, and it can be retrieved afterward
+// with Org.CseGetKubernetesClusterById and the returned ID.
+//
+// If the cluster is created correctly, returns all the available data in CseKubernetesCluster or an error if some of the fields
+// of the created cluster cannot be calculated or retrieved.
+func (org *Org) CseCreateKubernetesCluster(clusterData CseClusterSettings, timeoutMinutes time.Duration) (*CseKubernetesCluster, error) {
```

Comment:
```suggestion
func (org *Org) CseCreateKubernetesCluster(clusterData CseClusterSettings, timeout time.Duration) (*CseKubernetesCluster, error) {
```
If the timeout is of type `time.Duration`, there is no need to indicate the unit of measurement.

Reply:
Good catch, this started as an `timeoutMinutes int` and the name survived among refactors 😄